scripts/workers/worker_utils.py

```diff
@@ -96,3 +191,93 @@ def update_submission_status(self, data, challenge_pk):
         response = self.make_request(url, "PATCH", data=data)
         return response
+
+
+def download_and_extract_file(url, download_location):
+    """
+        * Function to extract download a file.
+        * `download_location` should include name of file as well.
+    """
+    try:
+        response = requests.get(url, stream=True)
+    except Exception as e:
+        logger.error("Failed to fetch file from {}, error {}".format(url, e))
+        traceback.print_exc()
+        response = None
+
+    if response and response.status_code == 200:
+        with open(download_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+
+
+def extract_zip_file(download_location, extract_location):
+    """
+    Helper function to extract zip file
+    Params:
+        * `download_location`: Location of zip file
+        * `extract_location`: Location of directory for extracted file
+    """
+    zip_ref = zipfile.ZipFile(download_location, "r")
+    zip_ref.extractall(extract_location)
+    zip_ref.close()
+
+
+def delete_zip_file(download_location):
+    """
+    Helper function to remove zip file from location `download_location`
+    Params:
+        * `download_location`: Location of file to be removed.
+    """
+    try:
+        os.remove(download_location)
+    except Exception as e:
+        logger.error(
+            "Failed to remove zip file {}, error {}".format(
+                download_location, e
+            )
+        )
+        traceback.print_exc()
+
+
+def download_and_extract_zip_file(url, download_location, extract_location):
+    """
+        * Function to extract download a zip file, extract it and then removes the zip file.
+        * `download_location` should include name of file as well.
+    """
+    try:
+        response = requests.get(url, stream=True)
+    except Exception as e:
+        logger.error("Failed to fetch file from {}, error {}".format(url, e))
+        response = None
+
+    if response and response.status_code == 200:
+        with open(download_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+        # extract zip file
+        extract_zip_file(download_location, extract_location)
+        # delete zip file
+        delete_zip_file(download_location)
+
+
+def create_dir(directory):
+    """
+        Creates a directory if it does not exists
+    """
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def create_dir_as_python_package(directory):
+    """
+        Create a directory and then makes it a python
+        package by creating `__init__.py` file.
+    """
+    create_dir(directory)
+    init_file_path = join(directory, "__init__.py")
+    with open(init_file_path, "w") as init_file:  # noqa
+        # to create empty file
+        pass
```

Comment:
@pushkalkatara I need your suggestion on this: Should you encapsulate this as a class `SubmissionFileHandler` or something on those lines.

CC: @RishabhJain2018

Reply:
Yes, that is a good idea. Thanks for the suggestion. I'll update the changes.

Reply:
Maybe we can rename it as `FileHandler` as the utils are being used for challenge and phase data creation as well.

Reply:
Yes, that's a good idea.

Reply:
`FileHandler` is better

Reply:
Fixed with the recent commit.